modules/@apostrophecms/schema/ui/apos/components/AposAttachment.vue

```diff
@@ -50,19 +56,78 @@ export default {
     },
     disabled () {
       return (this.limitReached || this.field.disabled);
+    },
+    moduleOptions() {
+      return window.apos.modules['@apostrophecms/file'];
     }
   },
   methods: {
     updated(items) {
       this.next = items;
     },
     validate(value) {
-      if (this.field.required && !value.data.length) {
+      if (this.field.required && !value) {
         return 'required';
       }
 
       return false;
-    }
+    },
+    async uploadMedia (event) {
+      this.$emit('upload-started');
+      const file = event.target.files[0];
+
+      const emptyDoc = await apos.http.post(this.moduleOptions.action, {
+        body: {
+          _newInstance: true
+        }
+      });
+      await apos.notify('Uploading file', { dismiss: true });
+
+      const formData = new window.FormData();
+
+      formData.append('file', file);
+      let attachment;
+
+      try {
+        attachment = await apos.http.post('/api/v1/@apostrophecms/attachment/upload', {
+          body: formData
+        });
+      } catch (error) {
+        console.error('Error uploading file.', error);
+
+        const msg = error.body && error.body.message ? error.body.message : 'Upload error';
+
+        await apos.notify(msg, {
+          type: 'danger',
+          icon: 'alert-circle-icon',
+          dismiss: true
+        });
+        return;
+      }
+
+      const fileData = Object.assign(emptyDoc, {
+        title: attachment.title,
+        attachment
+      });
+
+      try {
+        const uploaded = await apos.http.post(this.moduleOptions.action, {
+          body: fileData
+        });
+        await apos.notify('Upload Successful', {
```

Comment:
'Upload Successful' becomes 'Successfully uploaded FILENAME.EXT' with same truncation rules as before. Use the double-check icon as well